spec/stringProcessing/urlStructureSpec.js

```diff
@@ -0,0 +1,78 @@
+let UrlStructure = require( "../../js/stringProcessing/urlStructure" );
+
+describe( "UrlStructure", () => {
+	it( "accepts a url structure", () => {
+		let expected = "http://example.org/structure/";
+		let input = expected;
+
+		let urlStructure = new UrlStructure( input );
+
+		expect( urlStructure.getStructure() ).toBe( expected )
+	});
+
+	describe( "parseUrl", () => {
+		it( "parses a URL and builds a data structure from it", () => {
+			let expected = [
+				{
+					type: "static",
+					value: "http://example.org/",
+				},
+				{
+					type: "variable",
+					value: "%%placeholder%%",
+					name: "placeholder",
+				},
+				{
+					type: "static",
+					value: "/staticstuff/",
+				},
+				{
+					type: "variable",
+					value: "%%another%%",
+					name: "another",
+				},
+			];
+			let input = "http://example.org/%%placeholder%%/staticstuff/%%another%%";
+
+			let urlStructure = UrlStructure.fromUrl( input );
+
+			expect( urlStructure.getStructure() ).toEqual( expected );
+		} );
+
+		it( "parses an empty URL as an empty data structure", () => {
+			let expected = [];
+			let input = "";
+
+			let urlStructure = UrlStructure.fromUrl( input );
+
+			expect( urlStructure.getStructure() ).toEqual( expected );
+		} );
+	} );
+
+	describe( "buildUrl", () => {
```

Comment:
How should the function behave when inserting weird data like symbols or html entities? Maybe some tests for these cases could add some value?